database_table.sql

```diff
@@ -0,0 +1,112 @@
+CREATE TABLE organizations (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Members table
+CREATE TABLE members (
+  id SERIAL PRIMARY KEY,
+  username VARCHAR(255) NOT NULL,
+  email VARCHAR(255) NOT NULL,
+  password VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Organization members table
+CREATE TYPE roles as ENUM('organization_head', 'member');
+CREATE TABLE organization_members (
+  organization_id INT REFERENCES organizations(id),
+  member_id INT REFERENCES members(id),
+  organization_member_role roles ,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+-- Plans table
+CREATE TABLE plans (
+  id SERIAL PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  duration INT NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Organization plans table
+CREATE TABLE organization_plans (
+  organization_id INT REFERENCES organizations(id),
+  plan_id INT REFERENCES plans(id),
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Channels table
+CREATE TYPE channel_type as ENUM('Personal', 'Group');
+CREATE TABLE channels (
+  id SERIAL PRIMARY KEY,
+  channel_type channel_type;
+  name VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+-- Organization_Channel
+CREATE TABLE organization_channel(
+    channel_id INT REFERENCES channels(id),
+    organization_id INT REFERENCES organizations(id),
+);
+-- Channel members table
+CREATE TYPE statuses as ENUM('active', 'inactive');
+CREATE TABLE channel_members (
+  channel_id INT REFERENCES channels(id),
+  member_id INT REFERENCES members(id),
+  status statuses,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Messages table
+CREATE TABLE messages (
+  id SERIAL PRIMARY KEY,
+  channel_id INT REFERENCES channels(id),
+  member_id INT REFERENCES members(id),
+  message TEXT NOT NULL,
+  is_starred BOOLEAN NOT NULL DEFAULT false,
+  created_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  updated_at TIMESTAMP NOT NULL DEFAULT NOW()
+);
+
+-- Message read table
+CREATE TABLE message_read (
```

Comment:
why separate table? Can this not be part of messages table?

Reply:
The reasoning behind this was that retaining the message status data in the column would cause the data size to expand significantly as we had more data in the tables. Consequently, creating a new table aids in keeping the data more structured and speeds up query execution.